### DISCUSSION TITLE
Compiling problems: error C2011

### DISCUSSION THREAD

**Discussion by Belarusan** [hzqst/MetaHookSv#435] [General]
Hello. I want to put CaptionMod on Half-Life 25th Update.

I downloaded MetaHookSv-windows-x86.zip, and Windows Security blocked the archive, saying that the CaptionMod.dll file contains Trojan:Win32/Phonzy.B!m

So I'm trying to compile it myself (I don't see any other way).

I have Visual Studio 2022 with vc143, I have CMake, I have Git for Windows.

During compilation (scripts\build-MetaHook.bat) I have a failure:
```
Build FAILED.

"C:\Projects\GoldSource\MetaHook_build\MetaHookSv\MetaHook.sln" (MetaHook target) (1) ->
"C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj" (default target) (2) ->
(ClCompile target) ->
  C:\Projects\GoldSource\MetaHook_build\MetaHookSv\thirdparty\MINT\MINT.h(15980,1): error C2011: '_QUEUE_USER_APC_FLAGS
': 'enum' type redefinition [C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj]
  C:\Projects\GoldSource\MetaHook_build\MetaHookSv\thirdparty\MINT\MINT.h(15980,1): error C2011: '_QUEUE_USER_APC_FLAGS
': 'enum' type redefinition [C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj]
  C:\Projects\GoldSource\MetaHook_build\MetaHookSv\thirdparty\MINT\MINT.h(15980,1): error C2011: '_QUEUE_USER_APC_FLAGS
': 'enum' type redefinition [C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj]
```

Google doesn't help. I don't understand this at all. I'm embarrassed to ask here, but what am I doing wrong?

**Comment by hzqst** [comment]
The dependency "MINT" has been removed,  can you try again with the the latest commit?

**Reply by Belarusan** [reply]
Thank you. Now I got this:
```
Build FAILED.

"C:\Projects\GoldSource\MetaHook_build\MetaHookSv\MetaHook.sln" (MetaHook target) (1) ->
"C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj" (default target) (2) ->
(PreBuildEvent target) ->
  EXEC : CMake warning :  [C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj]


"C:\Projects\GoldSource\MetaHook_build\MetaHookSv\MetaHook.sln" (MetaHook target) (1) ->
"C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj" (default target) (2) ->
(ClCompile target) ->
  C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\LoadDllNotification.cpp(1,10): error C1083: Cannot open include
file: 'MINT.h': No such file or directory [C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj]
  C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\LoadDllNotification.h(3,10): error C1083: Cannot open include fi
le: 'MINT.h': No such file or directory [C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj]
  C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\LoadDllNotification.h(3,10): error C1083: Cannot open include fi
le: 'MINT.h': No such file or directory [C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj]

    1 Warning(s)
    3 Error(s)
```
I still think that I did something wrong…

**Reply by hzqst** [reply]
my bad. I've pushed the missing files to the latest commit. can you try again?

**Reply by Belarusan** [reply]
```
Build FAILED.

"C:\Projects\GoldSource\MetaHook_build\MetaHookSv\MetaHook.sln" (MetaHook target) (1) ->
"C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj" (default target) (2) ->
(ClCompile target) ->
  C:\Projects\GoldSource\MetaHook_build\MetaHookSv\thirdparty\MINT\include\MINT.h(15983,1): error C2011: '_QUEUE_USER_A
PC_FLAGS': 'enum' type redefinition [C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj]
  C:\Projects\GoldSource\MetaHook_build\MetaHookSv\thirdparty\MINT\include\MINT.h(15983,1): error C2011: '_QUEUE_USER_A
PC_FLAGS': 'enum' type redefinition [C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj]
  C:\Projects\GoldSource\MetaHook_build\MetaHookSv\thirdparty\MINT\include\MINT.h(15983,1): error C2011: '_QUEUE_USER_A
PC_FLAGS': 'enum' type redefinition [C:\Projects\GoldSource\MetaHook_build\MetaHookSv\src\MetaHook.vcxproj]
```

**Reply by hzqst** [reply]
Probably because you've installed an out-dated version of Windows SDK. You can fix the problem manually by removing

```
typedef enum _QUEUE_USER_APC_FLAGS
{
    QUEUE_USER_APC_FLAGS_NONE = 0x0,
    QUEUE_USER_APC_FLAGS_SPECIAL_USER_APC = 0x1,
} QUEUE_USER_APC_FLAGS;
```

at `/thirdparty/MINT/include/MINT.h` [@ line 15982](https://github.com/hzqst/MetaHookSv/blob/79618af638aaf62e6b6a4db780ebaaa0deeb0d03/thirdparty/MINT/include/MINT.h#L15982C1-L15986C24)

**Reply by Belarusan** [reply]
I apologize. I'm a novice at this. I figured since I have Windows 10, the SDK should be for Windows 10. But should I have chosen the Windows 11 SDK?

However, after removing those lines, I managed to compile the .exe and .dll files. And even managed to run in the game! It was really hard to figure out how, but I did it. I used the _svencoop_ folder from the _Build_ folder. And added files from _svencoop_hidipi_, because my monitor is 21:9 x34. And everything works perfectly.

Thank you so much for your help and for such a wonderful mod ❤️

***

Why is such a useful thing little known? I think a lot of people would like to play the original Half-Life with subtitles in their language.

In theory, do I have the right to translate the subtitles into my native language and post it as a mod? Of course, with the indication that this is MetaHook, with all original credits.

**Reply by hzqst** [reply]
> In theory, do I have the right to translate the subtitles into my native language and post it as a mod? Of course, with the indication that this is MetaHook, with all original credits.

as you wish

**Comment by i-s-a-a-m** [comment]
oh-
